core/bug_api.php

```diff
@@ -985,10 +985,18 @@ function bug_text_clear_cache( $p_bug_id = null ) {
  * @access public
  */
 function bug_exists( $p_bug_id ) {
-	if( false == bug_cache_row( $p_bug_id, false ) ) {
+	$c_bug_id = (int)$p_bug_id;
+
+	# Prevent db query errors
+	if( $c_bug_id > db_max_uint() ) {
```

Comment:
I thought of that. My reasoning is:
bug_cache_row doesnt perform input validation. The main use for that function is for known bug ids, and adding additional checks would add logic unneeded (by this premise). 
Note that this function expects an integer (i'd add, expects an existing bug id, within reasonable circumstances)), due to php loose types, that cannot be enforced by declaration.
 
bug_exists() is closer to the input layer, where validation should be performed, and is a good compromise for those pages that take arbitrary input. For example: view page relies on ensure_bug_exists for check the inputted bug_id, which then calls to bug_exists.

Notice how in bug_exists(), i added a cast to (int) for $p_bug_id, so if a garbage string is passed in, at least it's casted to 0 and returns no rows.